src/garage/tf/embeddings/encoder.py

```diff
@@ -8,6 +8,52 @@
 class Encoder(BaseEncoder, Module):
     """Base class for encoders in TensorFlow."""
 
+    def forward_n(self, input_values):
```

Comment:
forward is always vectorized, right?

Reply:
The non-vectorized version is used in `policy.get_latent()`, and `get_latent()` is used in sampler where a latent is sampled once per trajectory.

I make it this way also to keep consistent with the policy interface - it has `get_action()` as well as `get_actions()`.

Reply:
by your own analogy, shouldn't the name of these be get_latent() and get_latents()? Forward has a specific meaning in torch. In garage.tf.Module, the approximate equivalent is `build`

Reply:
`get_latent()` is in the policy while `forward()` is in the encoder. The first one makes a call to the second one.  `build()` assumes symbolic input. But in this case, both take actual values as input.

Maybe it is more appropriate to rename `forward()` to `get_latent()`? (The same name as it is in policies)

Reply:
I think get_latent is better